doc/lspconfig.txt

```diff
@@ -1,242 +1,639 @@
+
+- call `setup()` to enable the FileType autocmd
+>
+   lspconfig.foo_lsp.setup{}
+<
 ==============================================================================
-9. Built-in commands                             *lspconfig-built-in-commands*
+INDEX                                                          *lspconfig-index*
 
+The autogenerated documentation for all language servers is accessible in
+`server_configurations.md`. Type `gf` on the filename from this document to jump
+to the markdown file.
 
-* `:LspInfo` shows the status of active and configured language servers.
-The following support tab-completion for all arguments:
-* `:LspStart <config_name>` Start the requested server name. Will only successfully start if the command detects a root directory matching the current config. Pass `autostart = false` to your `.setup{}` call for a language server if you would like to launch clients solely with this command. Defaults to all servers matching current buffer filetype.
-* `:LspStop <client_id>` Defaults to stopping all buffer clients.
-* `:LspRestart <client_id>` Defaults to restarting all buffer clients.
 ==============================================================================
-10. The wiki                                              *lspconfig-the-wiki*
+ROOT DETECTION                                        *lspconfig-root-detection*
+                                                            *lspconfig-root-dir*
+
+lspconfig automatically launches language servers by defining a filetype
+autocommand based on the `filetypes` specified in the default configuration of
+each server, optionally overridable by the `filetypes` table passed to
+`setup`.
 
+This autocommand triggers a search from the current file position in the
+filesystem hierarchy up to the top level directory of your filesystem. The
+`root_dir` entry of each configuration is a function that returns true if the
+current directory in this traversal matches a given root pattern.
 
-Please see the [wiki](https://github.com/neovim/nvim-lspconfig/wiki) for
-additional topics, including:
-* [Installing language servers automatically](https://github.com/neovim/nvim-lspconfig/wiki/Installing-language-servers-automatically)
-* [Snippets support](https://github.com/neovim/nvim-lspconfig/wiki/Snippets)
-* [Project local settings](https://github.com/neovim/nvim-lspconfig/wiki/Project-local-settings)
-* [Recommended plugins for enhanced language server features](https://github.com/neovim/nvim-lspconfig/wiki/Language-specific-plugins)
+The following utility functions are provided by lspconfig. Each function call
+below, returns a function that takes as its argument the current buffer path.
+
+Examples are provided for didactic purpose:
+
+- `util.root_pattern`: function which takes multiple arguments, each 
+  corresponding to a different root pattern against which the contents of the
+  current directory are matched using |vim.fin.glob()| while traversing up the
+  filesystem.
```

Comment:
...and add "Example:" here.

(The other functions are not "examples" but _the_ correct usage.)